### ISSUE TITLE
pytorch 0.4.1 problems

### ISSUE THREAD

**Issue by ShigekiKarita** [espnet/espnet#320]
I'm testing 0.4.1 pytorch locally

https://github.com/pytorch/pytorch/releases

- no longer `_assert_no_grad` in torch
/src/nets/e2e_asr_attctc_th.py", line 462, in chainer_like_ctc_loss 
from torch.nn.modules.loss import _assert_no_grad

- forked warp-ctc causes segmentation faults even with its `tests/test_cpu.py`
  - I confirmed origin https://github.com/SeanNaren/warp-ctc was OK. maybe it is time to go back
  - in SeanNaren branch, there already has been `size_average` option that is same to our normalization https://github.com/SeanNaren/warp-ctc/blob/pytorch_bindings/pytorch_binding/warpctc_pytorch/__init__.py#L41

**Comment by ShigekiKarita** [comment]
duplicated #316
